tests/console/snapper_cleanup.pm

```diff
@@ -73,17 +73,53 @@ sub run {
         assert_script_run "snapper set-config NUMBER_LIMIT=2-10 NUMBER_LIMIT_IMPORTANT=4-10 SPACE_LIMIT=0.5";
     }
 
+    my ($n_scratch_prepost, $scratchfile_mb,          $safety_margin_mb,   $initially_free);
+    my ($nmax_snapshots,    $number_limit_upper_orig, $number_limit_upper, $n);
+
+    # three test parameters; hardcoded values found appropriate in past tests
+    $n_scratch_prepost = 3;
+    $scratchfile_mb    = 1024;
+    # Expected to keep the test from filling the last 300 MB of the filesystem, poo#36838
+    $safety_margin_mb = 300 + $scratchfile_mb;
+
     assert_script_run("snapper get-config");    # get initial cfg
+    $number_limit_upper_orig = script_output("snapper get-config | awk '/^NUMBER_LIMIT\\>/{ print \$3 }'");
     assert_script_run("snapper list");          # get initial list of snap's
-    assert_script_run("snapper set-config NUMBER_MIN_AGE=0");
+
+    # check amount of free fs disk space and adapt to it
+    $initially_free = get_space("$btrfs_fs_usage | awk -F ' ' '/Free .estimated.:.*min:/{print\$3}'");    # bytes
+    $nmax_snapshots = int(($initially_free / 1_048_576 - $safety_margin_mb) / $scratchfile_mb) * 2;
```

Comment:
Where does the magic number `1_048_576` come from?

Reply:
Come on, it's one gig 1024*1024 =D I wanted to comment about magic numbers, but here it's ok

Reply:
I need to transform from bytes to MBytes. 1048576 is 2^^20 or 1<<20 or 0x100000... If you prefer another way to code this constant as better readable, I'll be glad to oblige.

Reply:
well, we could use [Number::Bytes::Human](http://search.cpan.org/~ferreira/Number-Bytes-Human-0.11/Human.pm) after submitting to openSUSE:Factory and devel:openQA ...

ok, fine for me as is